components/brave_wallet/browser/keyring_controller.cc

```diff
@@ -193,17 +193,30 @@ void KeyringController::SetPrefForKeyring(PrefService* prefs,
 }
 
 // static
-void KeyringController::SetAccountNameForKeyring(
+void KeyringController::SetAccountInfoForKeyring(
     PrefService* prefs,
     const std::string& account_path,
-    const std::string& name,
+    const absl::optional<std::string> name,
+    const absl::optional<std::string> address,
     const std::string& id) {
+  if (!name && !address)
+    return;
   base::Value account_metas(base::Value::Type::DICTIONARY);
   const base::Value* value = GetPrefForKeyring(prefs, kAccountMetas, id);
   if (value)
     account_metas = value->Clone();
   base::Value account_meta(base::Value::Type::DICTIONARY);
-  account_meta.SetStringKey(kAccountName, name);
+  if (name)
+    account_meta.SetStringKey(kAccountName, *name);
+  else
+    account_meta.SetStringKey(
+        kAccountName, GetAccountNameForKeyring(prefs, account_path, id));
+
+  if (address)
+    account_meta.SetStringKey(kAccountAddress, *address);
+  else
+    account_meta.SetStringKey(
+        kAccountName, GetAccountAddressForKeyring(prefs, account_path, id));
```

Comment:
same here and also `kAccountName` here is wrong

Reply:
Fixed and rebased in 10bd683f95363d743b2fb72a09278f0b5c3bb92e